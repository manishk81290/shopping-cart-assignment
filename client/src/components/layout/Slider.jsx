import React from "react";
import { Row, Col, Carousel, Image } from "react-bootstrap";

const Slider = ({ banners }) => {
  return (
    <Row md={12} className="mt-100- slider">
      <Col>
        <Carousel interval={1000} className="bottom-shadow">
          {banners.map((banner) => {
            return (
              <Carousel.Item key={banner.id}>
                <Image
                  src={`${process.env.PUBLIC_URL}${banner.bannerImageUrl}`}
                  alt={banner.bannerImageAlt}
                />
              </Carousel.Item>
            );
          })}
          {/* <Carousel.Item>
            <img className="d-block w-100" src={OfferOne} alt="First slide" />
          </Carousel.Item>
          <Carousel.Item>
            <img className="d-block w-100" src={OfferTwo} alt="Second slide" />
          </Carousel.Item>
          <Carousel.Item>
            <img className="d-block w-100" src={OfferThree} alt="Third slide" />
          </Carousel.Item>
          <Carousel.Item>
            <img className="d-block w-100" src={OfferFour} alt="Third slide" />
          </Carousel.Item>
          <Carousel.Item>
            <img className="d-block w-100" src={OfferFive} alt="Third slide" />
          </Carousel.Item> */}
        </Carousel>
      </Col>
    </Row>
  );
};

export default Slider;
