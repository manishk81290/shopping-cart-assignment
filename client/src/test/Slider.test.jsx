import React from "react";
import Slider from "./../components/layout/Slider";
import ShallowRenderer from "react-test-renderer/shallow";

let realUseContext;
let useContextMock;
const sampleContextData = {
  banners: [
    {
      bannerImageUrl: "/static/images/offers/offer1.jpg",
      bannerImageAlt: "Independence Day Deal - 25% off on shampoo",
      isActive: true,
      order: 1,
      id: "5b6c38156cb7d770b7010ccc",
    },
  ],
  getBanners: () => {},
};

beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

test("Slider with mock useContext hook", () => {
  useContextMock.mockReturnValue(sampleContextData);
  const element = new ShallowRenderer().render(<Slider />);
  expect(element).toBeTruthy();
  expect(element).toMatchSnapshot();
});
