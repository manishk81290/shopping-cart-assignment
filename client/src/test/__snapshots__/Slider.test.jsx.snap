// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider with mock useContext hook 1`] = `
<Row
  className="mt-100- slider"
  md={12}
  noGutters={false}
>
  <Col>
    <Carousel
      className="bottom-shadow"
      controls={true}
      defaultActiveIndex={0}
      fade={false}
      indicators={true}
      interval={1000}
      keyboard={true}
      nextIcon={
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
      }
      nextLabel="Next"
      pause="hover"
      prevIcon={
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
      }
      prevLabel="Previous"
      slide={true}
      touch={true}
      wrap={true}
    >
      <CarouselItem>
        <Image
          alt="Independence Day Deal - 25% off on shampoo"
          fluid={false}
          rounded={false}
          roundedCircle={false}
          src="/static/images/offers/offer1.jpg"
          thumbnail={false}
        />
      </CarouselItem>
    </Carousel>
  </Col>
</Row>
`;
